### PR TITLE
Series of fixes for statsplot

### PR DESCRIPTION
**Pull request by sssangha**
Supports series of fixes outlined in #202, namely:
- Rounding of the colorbar ticks to two decimal points for floating values (as opposed to integer heatmaps of station locations).
- Variogram range to be reported in km units. Currently crude factor of 1010km applied, to be updated in PR #203 
- Remove the wording heatmap from the colorbar title
- Change "sill" to "Variance" and "range" to "Range"; "latitude" to "Latitude"; "longitude" to "Longitude"
- Ability to improve the dpm for the Figures, now toggled with ```-figdpi``` option
- Ability to change units of loaded arrays.

### CODE REVIEW

File: tools/bin/raiderStats.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/data2/ssangha/conda_installation/stable_jan10_2020/ariatools_conda/envs/RAiDER/bin/python3
```
Comment (jlmaurer) [inline tools/bin/raiderStats.py:1]: Looks like you made a temporary change, just need to switch back.

**Review by jlmaurer — COMMENTED** [review]
Looks fine to me except for the one environment statement.

File: tools/RAiDER/statsPlot.py [hunk 2]
```diff
@@ -351,6 +369,11 @@ def resid(x, d, v, m):
                                    loss='soft_l1', f_scale=0.1,
                                    args=(d, v, model))
 
+        # convert range to specified output unit
+        # TODO temp fix first multiply by 1,010,000 m as crude conversion from degrees to km
+        res_robust.x[0] = res_robust.x[0]*1010000
+        res_robust.x[0] = convert_SI(res_robust.x[0], 'm', 'km')
+
         with warnings.catch_warnings():
```
Comment (dbekaert) [inline tools/RAiDER/statsPlot.py:377]: Probably the easiest would be that you just compute distance and multiply it there with 110km per degree so you do that change once.

File: tools/RAiDER/statsPlot.py [hunk 3]
```diff
@@ -527,17 +550,25 @@ def plot_variogram(self, gridID, timeslice, coords, workdir='./', d_test=None, v
...
             plt.axhline(y=res_robust[1], color='g',
                         linestyle='--', label='ɣ\u0332\u00b2({}\u00b2)'.format(self.unit))
             plt.axvline(x=res_robust[0], color='c',
                         linestyle='--', label='h\u0332(°)')
         if d_test is not None and v_test is not None:
+            # TODO temp fix first multiply by 1,010,000 m as crude conversion from degrees to km
+            d_test = [i*1010000 for i in d_test]
             plt.plot(d_test, v_test, 'r-', label='experimental fit')
-        plt.xlabel('Distance (°)')
+        plt.xlabel('Distance (km)')
```
Comment (dbekaert) [inline tools/RAiDER/statsPlot.py:571]: Then in future when a user ask for a different unit. We would change units from SI to requested unit just before making the plot.

i.e. assume all units up to just before plotting to be in SI standard units. 
Then change it just before calling the plotting.

File: tools/RAiDER/statsPlot.py [hunk 4]
```diff
@@ -351,6 +369,11 @@ def resid(x, d, v, m):
                                    loss='soft_l1', f_scale=0.1,
                                    args=(d, v, model))
 
+        # convert range to specified output unit
+        # TODO temp fix first multiply by 1,010,000 m as crude conversion from degrees to km
+        res_robust.x[0] = res_robust.x[0]*1010000
```
Comment (dbekaert) [inline tools/RAiDER/statsPlot.py:374]: Note that equator 1 degree is roughly 110km so should be * 110,000 to go from degree to m and not 1,010,000.

File: tools/RAiDER/statsPlot.py [hunk 5]
```diff
@@ -351,6 +369,11 @@ def resid(x, d, v, m):
                                    loss='soft_l1', f_scale=0.1,
                                    args=(d, v, model))
 
+        # convert range to specified output unit
+        # TODO temp fix first multiply by 1,010,000 m as crude conversion from degrees to km
```
Comment (dbekaert) [inline tools/RAiDER/statsPlot.py:373]: the comment is incorrect it is mixing conversion from degree to m and km.

**Comment by dbekaert** [comment]
@sssangha can you also add in an exponential model without sill?

File: tools/bin/raiderStats.py (hunk 1 above)
Comment (sssangha) [reply tools/bin/raiderStats.py]: Ok, fixed.

File: tools/RAiDER/statsPlot.py (hunk 2 above)
Comment (sssangha) [reply tools/RAiDER/statsPlot.py]: Figured out how to change the distance once. The bug I experience yesterday is gone.

File: tools/RAiDER/statsPlot.py (hunk 4 above)
Comment (sssangha) [reply tools/RAiDER/statsPlot.py]: Fixed

File: tools/RAiDER/statsPlot.py (hunk 5 above)
Comment (sssangha) [reply tools/RAiDER/statsPlot.py]: Fixed

File: tools/RAiDER/statsPlot.py [hunk 6]
```diff
@@ -369,7 +389,10 @@ def __exponential__(self, parms, h):
         a, b, c = parms
```
Comment (jlmaurer) [inline tools/RAiDER/statsPlot.py:389]: With nugget = False will this fail?

**Comment by sssangha** [comment]
> @sssangha can you also add in an exponential model without sill?

You mean no nugget? By default I turned off the nugget flag in the exponential model. Will add in user-flag to toggle later.

**Review by jlmaurer — COMMENTED** [review]
Just one place where the nugget = False could cause a problem.

File: tools/RAiDER/statsPlot.py (hunk 6 above)
Comment (jlmaurer) [reply tools/RAiDER/statsPlot.py:389]: I think I've convinced myself it should be ok.

**Review by jlmaurer — APPROVED** [review]

**Review by dbekaert — APPROVED** [review]
LGTM @sssangha

**Pull request merged**